src/collective/collectionfilter/filteritems.py

```diff
@@ -9,7 +9,12 @@
 from collective.collectionfilter.vocabularies import DEFAULT_FILTER_TYPE
 from collective.collectionfilter.vocabularies import EMPTY_MARKER
 from Missing import Missing
+from plone.app.blocks.layoutbehavior import ILayoutBehaviorAdaptable
 from plone.app.contenttypes.behaviors.collection import ICollection
+try:
+    from plone.app.blocks.layoutbehavior import ILayoutAware
+except ImportError:
+    ILayoutAware = None
```

Comment:
Instead of try/except imports I'd refactor the whole `ICollectionish` logic into the tiles folder which is included onyl if `plone.tiles` is present ... collective.collectionfilter doesn't (shouldn't) have a hard dependency on plone.app.mosaic.

Reply:
it's not a hard dependency

Reply:
I don't think it makes the code more readable to split it up between tiles folder and filtereditems but I've done it. esp since all the GEO stuff has been kept togeather yet its not put in its own directory.